GameDataAPI promise / error types

The issue comes from `ts-lint` insisting on explicit typing, forcing use of `:any` on the value argument as in

```
webAPI.GameDataAPI.getRacesV1().then((value: any) => dispatch(something));
```
rather than just
```
webAPI.GameDataAPI.getRacesV1().then((value) => dispatch(something));
```

The promise argument type is 

```
{
  duration: number;
  problem: any;
  ok: boolean;
  status: number;
  headers: any;
  config: AxiosRequestConfig;
  data: any;
  error: Unauthorized | BadRequest | ExecutionError | NotAllowed | ServiceUnavailable
}
```

Also because error may be one of several types, I wonder if it would be better if they were all derived from a base error type

```
export interface RequestError {
  Code: number;
  Message: string;
  FieldCodes: any;
};

export interface BadRequest extends RequestError {
  Code: 500;
  Message: string;
  FieldCodes: [ FieldCodes.UnexpecifiedExceptionError ]
};
```  

And then we can use `error: RequestError` when passing `value.error` around.

OR, we could just disable `ts-lint`'s insistence of having explicitly defined types.

The actual error ts-lint was throwing is:

``Unexpected parentheses around single function argument having a body with no curly braces.``
for
```.then((value) => dispatch(something))```

and doing the following works
```.then((value: any) => dispatch(something))```

seems also doing the following works

```.
then((value) => {
  dispatch(something)
})
```
As for the error argument, typescript complains that it implicitly has any type, I guess because error has more than one possible type.

``error TS7006: Parameter 'error' implicitly has an 'any' type.``

Anyway, so I can use { } in arrow functions to avoid the weird ts-lint error instead of adding :any to value argument, but for now error: any will need to stay I think.

--- a/patcher/src/widgets/CharacterCreation/services/session/attributeOffsets.ts
+++ b/patcher/src/widgets/CharacterCreation/services/session/attributeOffsets.ts
@@ -7,9 +7,9 @@
 import { Promise } from 'es6-promise';
 import 'isomorphic-fetch';
 import { Race, Gender, webAPI } from 'camelot-unchained';
+type CSEError = webAPI.Errors.CSEError;
 
 import { fetchJSON } from '../../lib/fetchHelpers';
-import ResponseError from '../../lib/ResponseError';
 
 export interface AttributeOffsetInfo {
   race: Race;
@@ -43,7 +43,7 @@ export function fetchAttributeOffsetsSuccess(offsets: AttributeOffsetInfo[]) {
   };
 }
 
-export function fetchAttributeOffsetsFailed(error: any) {
+export function fetchAttributeOffsetsFailed(error: CSEError) {
   return {
     type: FETCH_ATTRIBUTE_OFFSETS_FAILED,
     error: error.Message,
@@ -54,7 +54,7 @@ export function fetchAttributeOffsets(shard: number = 1) {
   return (dispatch: (action: any) => any) => {
     dispatch(requestAttributeOffsets());
     return webAPI.GameDataAPI.getAttributeOffsetsV1(shard)
-      .then((value: any) => {
+      .then((value) => {
         dispatch(value.ok
                   ? fetchAttributeOffsetsSuccess(value.data)
                   : fetchAttributeOffsetsFailed(value.error));

--- a/patcher/src/widgets/CharacterCreation/services/session/attributes.ts
+++ b/patcher/src/widgets/CharacterCreation/services/session/attributes.ts
@@ -6,10 +6,10 @@
 
 import {Promise} from 'es6-promise';
 import 'isomorphic-fetch';
-import {Race, Gender, webAPI} from 'camelot-unchained';
+import {Race, Gender, webAPI } from 'camelot-unchained';
+type CSEError = webAPI.Errors.CSEError;
 
 import {fetchJSON} from '../../lib/fetchHelpers';
-import ResponseError from '../../lib/ResponseError';
 
 const totalPoints = 30;
 
@@ -73,10 +73,10 @@ export function fetchAttributesSuccess(attributes: AttributeInfo[]) {
   };
 }
 
-export function fetchAttributesFailed(error: ResponseError) {
+export function fetchAttributesFailed(error: CSEError) {
   return {
     type: FETCH_ATTRIBUTES_FAILED,
-    error: error.message,
+    error: error.Message,
   };
 }
 
@@ -84,7 +84,7 @@ export function fetchAttributes(shard: number = 1) {
   return (dispatch: (action: any) => any) => {
     dispatch(requestAttributes());
     return webAPI.GameDataAPI.getAttributeInfoV1(shard)
-      .then((value: any) => {
+      .then((value) => {
         if (value.ok) {
           value.data.map((a: any) => {
             a.allocatedPoints = 0;

--- a/patcher/src/widgets/CharacterCreation/services/session/factions.ts
+++ b/patcher/src/widgets/CharacterCreation/services/session/factions.ts
@@ -8,8 +8,8 @@ import { Promise } from 'es6-promise';
 import 'isomorphic-fetch';
 
 import { fetchJSON } from '../../lib/fetchHelpers';
-import ResponseError from '../../lib/ResponseError';
 import { webAPI } from 'camelot-unchained';
+type CSEError = webAPI.Errors.CSEError;
 
 export interface FactionInfo {
   id: number;
@@ -44,10 +44,10 @@ export function fetchFactionsSuccess(factions: FactionInfo[]) {
   };
 }
 
-export function fetchFactionsFailed(error: ResponseError) {
+export function fetchFactionsFailed(error: CSEError) {
   return {
     type: FETCH_FACTIONS_FAILED,
-    error: error.message,
+    error: error.Message,
   };
 }
 
@@ -62,7 +62,7 @@ export function fetchFactions(shard: number = 1) {
   return (dispatch: (action: any) => any) => {
     dispatch(requestFactions());
     return webAPI.GameDataAPI.getFactionInfoV1()
-      .then((value: any) => {
+      .then((value) => {
         dispatch(value.ok ? fetchFactionsSuccess(value.data) : fetchFactionsFailed(value.error));
       });
   };

--- a/patcher/src/widgets/CharacterCreation/services/session/races.ts
+++ b/patcher/src/widgets/CharacterCreation/services/session/races.ts
@@ -7,10 +7,9 @@
 import { Promise } from 'es6-promise';
 import 'isomorphic-fetch';
 import { Race, Faction, webAPI } from 'camelot-unchained';
+type CSEError = webAPI.Errors.CSEError;
 
 import { fetchJSON } from '../../lib/fetchHelpers';
-import ResponseError from '../../lib/ResponseError';
-
 
 export interface RaceInfo {
   name: string;
@@ -40,7 +39,7 @@ export function fetchRacesSuccess(races: RaceInfo[]) {
   };
 }
 
-export function fetchRacesFailed(error: any) {
+export function fetchRacesFailed(error: CSEError) {
   return {
     type: FETCH_RACES_FAILED,
     error: error.Message,
@@ -65,7 +64,7 @@ export function fetchRaces(shard: number = 1) {
   return (dispatch: (action: any) => any) => {
     dispatch(requestRaces());
     return webAPI.GameDataAPI.getRacesV1()
-      .then((value: any) => {
+      .then((value) => {
         dispatch(value.ok ? fetchRacesSuccess(value.data) : fetchRacesFailed(value.error));
       });
   };

